restler/engine/core/async_request_utilities.py

```diff
@@ -198,9 +199,16 @@ def try_async_poll(request_data, response, max_async_wait_time, poll_delete_stat
                                     responses_to_parse.insert(0, get_response)
                             # Break and return the responses to be parsed
                             break
+                        elif int(poll_response.status_code) >= 300:
+                            # Error obtaining the polling response.
+                            # It may still be possible to extract the relevant ID out of the response
+                            responses_to_parse.append(poll_response)
```

Comment:
poll_response in this case is the 404 not found. If we simply break and don't add responses_to_parse in this else if, the if at line 267 will automatically add the original response, which will bypass the 404 all together and resolve this issue.

Reply:
I see, thanks, I had mis-copied that from your logs.  I'll make the change shortly, please let me know when you've tried it to confirm so I can merge.